pyflakes/checker.py

```diff
@@ -1234,6 +1234,9 @@ def handleNodeLoad(self, node):
         if name == '__module__' and isinstance(self.scope, ClassScope):
             return
 
+        if not PY2 and name == '__qualname__' and isinstance(self.scope, ClassScope):
+            return
+
```

Comment:
another idea would be to compute this dynamically:

```python
>>> class C:
...     class_scoped_magic = dir()
... 
>>> C.class_scoped_magic
['__module__', '__qualname__']
```

this would be more future-proof -- what do you think?

Reply:
Good idea!